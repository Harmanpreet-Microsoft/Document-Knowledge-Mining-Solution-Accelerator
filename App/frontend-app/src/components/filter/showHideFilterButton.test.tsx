// src/components/filter/showHideFilterButton.test.tsx
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import {FilterButton} from './showHideFilterButton';

describe('FilterButton component', () => {
  it('renders the button with the correct text', () => {
    render(<FilterButton />);
    expect(screen.getByRole('button')).toHaveTextContent('components.filter.title'); // Note: This will be the key, not the actual translation
  });

  it('calls the onFilterPress callback when clicked', () => {
    const onFilterPress = jest.fn();
    const { getByRole } = render(<FilterButton onFilterPress={onFilterPress} />);
    const button = getByRole('button');
    fireEvent.click(button);
    expect(onFilterPress).toHaveBeenCalledTimes(1);
  });

  it('renders the button with the correct icon', () => {
    render(<FilterButton />);
    const button = screen.getByRole('button');
    expect(button).toContainElement(button.querySelector('svg'));
  });


  it('renders the button with the correct class name', () => {
    render(<FilterButton />);
    expect(screen.getByRole('button')).toHaveClass('h-full');
  });
});
