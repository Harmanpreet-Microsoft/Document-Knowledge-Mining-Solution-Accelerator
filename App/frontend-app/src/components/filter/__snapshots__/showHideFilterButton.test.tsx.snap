// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="fui-Button r1alrhcs h-full ___13arsol_wkbbhs0 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl"
    type="button"
  >
    <span
      class="fui-Button__icon rywnvv2 ___963sj20_1qpm2bl f1nizpg2"
    >
      <svg
        aria-hidden="true"
        class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.25 13.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5h4.5Zm2-4.25a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5h8.5Zm2-4.25a.75.75 0 0 1 0 1.5H3.75a.75.75 0 0 1 0-1.5h12.5Z"
          fill="currentColor"
        />
      </svg>
    </span>
    components.filter.title
  </button>
</DocumentFragment>
`;
